### DISCUSSION TITLE
Is EDC natively support ORDL policy ?

### DISCUSSION THREAD

**Discussion by scandinave** [eclipse-edc/Connector#4606] [Q&A]
Hi, I try to make a ForbidAll policy.

I have try to specify my policy like this : 

```json
{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@id": "identity-policy",
  "@type": "PolicyDefinition",
  "policy": {
    "@context": "http://www.w3.org/ns/odrl.jsonld",
    "@type": "Set",
    "conflict": "perm",
    "permission": [
      
    ],
    "prohibition": [
      {
        "action": "use",
        "target" : "*"
      }
    ],
    "obligation": []
  }
}
```

And associate it to assets via a contract definition as accessPolicy. When I try to create an offer from the 2nd connector, it's as the policy is not evaluated. So my questions are : 

*  Does EDC implement basic ODRL support outside of constraints ? 
* If no, is it possible to map this type of ODRL policy (without constraint) to a custom implementation ? If yes, how to do that ? Documentation only refers to a constraintPolicy.

Thx

**Comment by jimmarino** [comment]
Hi,

No, because ODRL is too open-ended. At the DSP specification level, we are working on more explicitly defining what ODRL expression types are supported (via Json Schema) and EDC will conform to that.

To understand what EDC currently supports, please refer to the documentation.

**Reply by scandinave** [reply]
I have already read the documentation.The only use case that seems to be supported is a policy with a Constraint. It does not make sens to add constraint for a ForbidAll policy. 

How can I bind the above definition to a Java class to make the proper evaluation?

**Reply by jimmarino** [reply]
Please make sure you have reviewed the documentation carefully, especially regarding policy scopes and bindings. You can also implement policy engine pre- and post-validator extensions, which are documented in the SPI and test cases.

**Reply by scandinave** [reply]
Maybe you can point me to the relevant part of the documentation. The search function on the doc website is completely broken.
